CHANGELOG.md

```diff
@@ -11,6 +11,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Added configurable logging macros to stm32 platform
 - Added support for ULP wakeup on ESP32
 - Added heap growth strategies as a fine-tuning option to `spawn_opt/2,4`
+- Added support for creations of binaries with unaligned strings
```

Comment:
Does this affect an of the caveats in the welcoming documentation?  I think we say explicitly that we don't support bit strings on unaligned bits.

https://github.com/atomvm/AtomVM/blob/master/doc/src/programmers-guide.md?plain=1#L58

Reply:
Good point.

Unfortunately, we still don't support bitstrings and most operations (this would be #154)

I only fixed a corner case of encoding of bits as in the test. Indeed, the following expression works with master and OTP21 but fails with OTP22 or higher:
```
Y1 = <<A:16/little, X1:7, 1:1, 11:4, 3:3, X2:1>>
```
The reason is the compiler transforms it in:
```
Y1 = <<A:16/little, X1:7, "Û", X2:1>>
```
So this PR adds support for the general case:
```
<<0:3, "hello", 0:5>>.
```
Do you think we should amend the guide? Who's doing this, except by accident as an optimization of the compiler?

Reply:
I wouldn't know what to say in the guide.  I suspect the caveat I have in there now might not be as precise as it could be (?)